src/TLuaInterpreter.cpp

```diff
@@ -3042,9 +3046,19 @@ int TLuaInterpreter::setBackgroundColor(lua_State* L)
             return lua_error(L);
         } else {
             r = static_cast<int>(lua_tonumber(L, s));
+
+            if (!validRange(r)) {
+                lua_pushfstring(L, "setBackgroundColor: bad argument #%d type (red value needs to be between 0-255, got %d!)", s, r);
```

Comment:
As I say - It is not a **bad type** (string|nil|integer|bool|table|some other lua objects) it is a bad *value* so this message is misleading, IMNSHO!

Reply:
Agree! Thanks, fixed.